src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -8,7 +8,9 @@ public class Query
     {
         public bool IsDistinct;
         public QueryExpression Where;
-        public (FieldExpression From, StringSegment? Alias, QueryExpression Filter, bool Index) From;
+        public FromClause From;
+        public List<WithClause> WithClauses;
+        public List<MatchClause> MatchClauses { get; set; }
```

Comment:
We will but they can all reside under MatchClause and then have MatchExpression because we will need to support things like (foo)-[:Like]->(bar) and (cat)-[:Hates]->(dogs)